aopy/analysis.py

```diff
@@ -1543,3 +1543,61 @@ def interp_multichannel(x):
     x[nan_idx] = np.interp(idx,xp,fp)
 
     return x
+
+
+'''
+Behavioral metrics 
+'''
+
+def compute_path_length_per_trial(trajectory):
+    '''
+    This function calculates the path length by computing the distance from all points in the trajectory. It returns a single value for path length.
+
+    Args:
+        trajectory (nt x 2): Input trajectory, could be a corsor trajectory or eye trajectory
+
+    Returns:
+        path_length (float): length of the trajectory
+    '''
+    lengths = np.sqrt(np.sum(np.diff(trajectory, axis=0)**2, axis=1)) # compute the distance from all points in trajectory
+    path_length = np.sum(lengths)
+    return path_length
+
+
+def time_to_target(event_codes, event_times, per_target_stats=False):
+    '''
+    This function calculates reach time to target only on rewarded trials given event codes and event times.
+
+    Args:
+        event_codes (list) : event codes
+        event_times (list) : event times corresponding to the event codes. These event codes and event times could be the output of preproc.base.get_trial_segments_and_times().
```

Comment:
are these lists of events or lists of lists of events, where each list is a trial? in other words, is the input to this function supposed to be trial separated already?

Reply:
Yes, input to this function is trail-aligned event codes. Since we have another function to trial align event codes and time. It could be done outside the function.

Reply:
ok, can you please make it more clear in the docstring what the inputs are supposed to be? or give an example in the docstring?

Reply:
Already updated function documentation in a commit yesterday to make it clear. Please check latest version and see if further explanation is required.